### ISSUE TITLE
Track IRC users by ident, not by nick

### ISSUE THREAD

**Issue by jansol** [matrix-org/matrix-appservice-irc#72]
IRC -> matrix user mapping should use the IRC user's ident (usually something like nick!user@host, which is also used for bans etc) instead of the current nick to identify IRC users. This is needed for #71 among other things.

**Comment by Kegsay** [comment]
Agreed, the bridge should be using that for the reasons you give (#71). This would require a fairly harsh rework of the internals though (most things are keyed off the nick I->M side) - but would need to be done the moment we want to add any ability for IRC users to be able to customise their Matrix user (e.g. display name).

**Comment by Mikaela** [comment]
nick!user@host is actually the hostmask, ident is just another word for username. 

The title confused me as I wasn't sure how would ident based tracking work.

**Comment by jansol** [comment]
My bad, fixed now.

**Comment by erdnaxeli** [comment]
If we do this, how can a matrix user could still talk to an IRC user? Today I can guess its MXID because it is build with its nick, but with its hostmask it would be quit impossible.

**Comment by Kegsay** [comment]
I don't know to be honest.

This is a common problem for bridges (e.g. random Slack IDs on the slack bridge) so I was planning on looking for prior art here.

**Comment by erdnaxeli** [comment]
Maybe something like `!query <network> <user>` would do the job?

**Comment by jansol** [comment]
> `!query <network> <user>`

That would work, also riot's suggestions based on display name are relatively usable nowadays. (for users that share a room with you)

**Comment by hifi** [comment]
IRC users cannot be tracked by partial hostmasks (user@host) as they are not unique and if you use the full hostmask (nick!user@host) it's practically the same as tracking the nick only.
